interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -255,9 +264,42 @@ namespace cling {
     const clang::ASTContext& ctx = getASTContext();
     unsigned payloadSize = ctx.getTypeSizeInChars(getType()).getQuantity();
     m_Storage.m_Ptr = AllocatedValue::CreatePayload(payloadSize, dtorFunc,
-                                                    GetNumberOfElements());
+                                                GetNumberOfElements(getType()));
+  }
+
+  void Value::AssertTypeMismatch(const char* Type) const {
+    assert(isBuiltinType() && "Must be a builtin!");
+    const clang::BuiltinType *BT = getType()->castAs<clang::BuiltinType>();
+    clang::PrintingPolicy Policy = getASTContext().getPrintingPolicy();
+    assert(BT->getName(Policy).equals(Type));
```

Comment:
```suggestion
    assert(BT->getName(getASTContext().getPrintingPolicy()).equals(getType()->castAs<clang::BuiltinType>()));
```

Reply:
I prefer the previous solution which does not violate the 80-col rule we have in cling.

Reply:
Fair enough, even-though it means that if the function is mis-used then it will cost instead of being costless (in a Release build).

Reply:
Resolved by hidding the declaration in the NDEBUG case.